pandas/core/config_init.py

```diff
@@ -341,8 +343,26 @@ def is_terminal() -> bool:
         validator=is_instance_factory([type(None), int]),
     )
     cf.register_option("max_categories", 8, pc_max_categories_doc, validator=is_int)
+
+    def _deprecate_negative_int_max_colwidth(key):
+        value = cf.get_option(key)
+        if value is not None and value < 0:
+            warnings.warn(
+                "Passing a negative integer is deprecated in version 1.0 and "
+                "will not be supported from version 1.2. Instead, use None to "
```

Comment:
just say future version, we are not sure when this will be removed at this piont.